### ISSUE TITLE
[Beta] Each time the n or p Hotkey is used, the editor refocuses on Windows

### ISSUE THREAD

**Issue by RandellDawson** [freeCodeCamp/freeCodeCamp#37146]
**Problem:** When tested on Chrome and Firefox (on Windows) and using the `p` or `n` hotkeys, we are supposed to be able to navigate to the next and previous lessons without the editor going back into focus while in the escaped mode. However, on the beta site, I have to keep escaping the editor every single time I click the `p` or `n` once I first escape the editor. The original navigation buttons that were removed in favor of the hot keys did not do this.

**Comment by raisedadead** [comment]
Thanks. Please share the detailed config for your OS.

**Comment by RandellDawson** [comment]
Windows 10 Home edition.  Using Chrome Version 77.0.3865.90 (Official Build) (64-bit) and Firefox 69.0.2 (64-bit).

**Comment by thecodingaviator** [comment]
I have been able to reproduce this. What is happening is every time `n` or `p` is pressed, the new challenge is opened and then the focus of the cursor moves to the editor which makes further navigation impossible unless someone uses their mouse to refocus somewhere else

**Comment by raisedadead** [comment]
Thanks, we know this is confirmed. Lets keep the thread for solutions.

**Comment by vkWeb** [comment]
@ojeytonwilliams Any thoughts on this?

**Comment by rickysyamsi** [comment]
Hi people, first time contributor here.

I have tested my fix locally with both latest chrome and firefox. Also there is something that I want to make sure. Are there any specific conditions when auto focus on editor is preferred rather than smooth navigation to previous and next lessons? If there are none, then this issue should be hopefully done.

**Comment by ojeytonwilliams** [comment]
@vkWeb I've not been able to reproduce this yet.  It works as intended (for me) on Chrome and Firefox on both Windows and Ubuntu, so if anyone could walk me through the steps, that would be a big help.

@rickysyamsi the original intention was that the editor is focused by default, since that creates the cleanest flow.  I.e. if the user wants to immediately get started using the editor, they can.  Pressing escape should allow them to navigate freely until they re-focus on the editor restoring the previous behaviour.

**Comment by rickysyamsi** [comment]
@ojeytonwilliams I understand, thank you for the explanation. I didn't know before that escape key can be used to quit focus from editor.

While we are talking about flow and keyboard navigation. I think that default auto focus should be on challenge description (left part) instead of editor (middle part).

I think that when students are accessing a challenge page, most of them will be reading the challenge description first. Then after finishing it, they should be able to access editor easily.

Currently when using keyboard navigation, students need to press multiple tabs to access editor. I think that it will be very helpful if students can access editor easily with single keyboard press (for example 'e' key).

I also just realize that I can not scroll challenge description part with keyboard navigation (up and down keys). Should I create a new issue for this?

**Comment by raisedadead** [comment]
The idea is to not need to touch your mouse more than its needed. 

So when you complete the challenge, your next one is served with the editor already in focus. 

The hotkeys for nav are really a nice to have and by no means our primary way of navigating.

To quit the focus on the editor and enter a nav mode, you hit escape and start pressing the hot key `p` or `n`.

The editor does not resume focus unless you click on it.

Once it is in focus, the nav is ended and you have to hit esc again to enter nav.

> I also just realize that I can not scroll challenge description part with keyboard navigation (up and down keys). Should I create a new issue for this?

Yes please.

**Comment by raisedadead** [comment]
As of this comment the issue is that on Windows at least as reported by OP, the editor automatically gets focused (our intended parent behaviour in non-navigation mode).

This is an annoying experience, nonetheless only for users using the nav mode, which we have not yet advertised.

So the fix would be to check and handle the use cases as mentioned in my last comment.

**Comment by rickysyamsi** [comment]
Now I understand the original intention. So normal non-navigation mode is auto focus on editor by default, then user can enter navigation mode by pressing escape key. It should remain in navigation mode until user click on the editor again. Then it will return back to normal non-navigation mode.

I will try to fix it again, thanks for the explanation.

**Comment by ojeytonwilliams** [comment]
> Currently when using keyboard navigation, students need to press multiple tabs to access editor. I think that it will be very helpful if students can access editor easily with single keyboard press (for example 'e' key).

@rickysyamsi That's a nice idea! Please could you add that to the up down arrow issue, so we can discuss adding that as a feature.

After chatting with @thecodingaviator I think I've pinned down how to reproduce the issue.  

1. Open any challenge
2. Click on the instructions
3. Press escape
4. Press n

The new challenge will open and the editor will get focused.

Pressing escape only works if the editor is already focused, but clearly that's not how people are using it in practice.  I think we should enter navigation mode whenever the Hotkeys have focus.

@rickysyamsi You're welcome to try and fix this, but don't feel obliged - I can take care of it tomorrow.

**Comment by raisedadead** [comment]
I would recommend @ojeytonwilliams to take a dig at this, probably would be way faster for you to resolve.

@rickysyamsi you can go ahead and create a pull request for the arrow nav.

Thanks you all.

**Comment by rickysyamsi** [comment]
I understand, @ojeytonwilliams please take over this issue. @raisedadead I will try to dig on the arrow nav issue then.

Thank you guys :)

**Comment by RandellDawson** [comment]
@ojeytonwilliams My steps to reproduce the problem are similar to yours, but I don't have to do your step 3.  I simply:
1. Open any challenge.
2. Click on the instructions.
3. Press `n` or `p`.

**Comment by thecodingaviator** [comment]
@RandellDawson both ways work since both are doing the exact same job i.e. taking the focus off the editor

**Comment by ojeytonwilliams** [comment]
The idea is that unless you actively escape, you aren't in navigation mode and it should behave normally.  So, if you're in the flow "Open challenge, read instructions, start typing, ctrl + enter, ctrl + enter, repeat" you never need to touch the mouse as the editor always takes focus when you enter a challenge.

If you press escape (currently only when in the editor.  After #37167 everywhere except the header and the preview) then the editor will not get focused automatically i.e. you're in navigation mode.

The reason for making people press escape is to prevent them accidentally switching to navigation mode just by clicking on, say, the instructions and being annoyed that they have to keep selecting the editor.  Not something they have to do at the moment.
